vnext/Microsoft.ReactNative.Cxx/JSValue.h

```diff
@@ -233,8 +233,8 @@ inline winrt::Windows::UI::Xaml::Media::Brush JSValue::To() const noexcept {
 
 template <class T>
 static JSValue From(const T &value) noexcept {
-  auto writer = MakeJSValueTreeWriter();
-  WriteValue(*writer, value);
+  auto const &writer = MakeJSValueTreeWriter();
```

Comment:
>auto const &writer [](start = 2, length = 18)

This is bad: keeping a reference to a temporary will cause accessing object that is already deleted.
Why to do this change?

Reply:
the object that's created ends its lifetime at the exit of the function so this seems safe. Why do you think this object will be deleted early? when?

---
In reply to: [379160102](https://github.com/microsoft/react-native-windows/pull/4101#discussion_r379160102) [](ancestors = 379160102)

Reply:
This was to fix the build break.

Reply:
@asklar object returned by MakeJSValueTreeWriter() will be destroyed right away and the writer variable will keep a reference to a deleted variable. It is one of the most common issues I see in code quite often because developers do not know about C++ object lifetime. You may not see the crash in Debug build, but it will be 100% repro in Release build. VS C++ compiler gives a warning about it. @jonthysell I want to know what type of build break was there. The code compiles just fine today. I use VS 2019 every day. While I understand the need to move to the new toolset, we must not introduce bugs like this one.

Reply:
Actually, it seems that C++ has an exception for const&: https://herbsutter.com/2008/01/01/gotw-88-a-candidate-for-the-most-important-const/
While I do not like the change and will revert back after the merge, I agree it is not that severe as I thought.

Reply:
> @asklar object returned by MakeJSValueTreeWriter() will be destroyed right away and the writer variable will keep a reference to a deleted variable. It is one of the most common issues I see in code quite often because developers do not know about C++ object lifetime. You may not see the crash in Debug build, but it will be 100% repro in Release build. VS C++ compiler gives a warning about it. @jonthysell I want to know what type of build break was there. The code compiles just fine today. I use VS 2019 every day. While I understand the need to move to the new toolset, we must not introduce bugs like this one.

Please compile the unit tests in win32, using VS 2019 and the v142 toolset, not v141.

Reply:
I see it with v142. The fix is very simple: we just need to remove the '*'. I have added the fix to my current PR [https://github.com/microsoft/react-native-windows/pull/4124](https://github.com/microsoft/react-native-windows/pull/4124)

---
In reply to: [380847489](https://github.com/microsoft/react-native-windows/pull/4101#discussion_r380847489) [](ancestors = 380847489)